src/core/format/RawAddress.ts

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2018 NEM
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { sha3_256 } from 'js-sha3';
+import RIPEMD160 = require('ripemd160');
+import { Base32 } from './Base32';
+import { Convert } from './Convert';
+import { RawArray } from './RawArray';
+
+export class RawAddress {
+    static readonly constants = {
+        sizes: {
+            ripemd160: 20,
+            addressDecoded: 25,
+            addressEncoded: 40,
+            key: 32,
+            checksum: 4,
+        },
+    };
+    /**
+     * Converts an encoded address string to a decoded address.
+     * @param {string} encoded The encoded address string.
+     * @returns {Uint8Array} The decoded address corresponding to the input.
+     */
+    public static stringToAddress = (encoded) => {
+        if (RawAddress.constants.sizes.addressEncoded !== encoded.length) {
+            throw Error(`${encoded} does not represent a valid encoded address`);
+        }
+
+        return Base32.Base32Decode(encoded);
+    }
+
+    /**
+     * Format a namespaceId *alias* into a valid recipient field value.
+     * @param {string} namespaceId The hexadecimal namespaceId
+     * @returns {Uint8Array} The padded hexadecimal notation of the alias
+     */
+    public static aliasToRecipient = (namespaceId) => {
+        // 0x91 | namespaceId on 8 bytes | 16 bytes 0-pad = 25 bytes
+        const padded = new Uint8Array(1 + 8 + 16);
+        padded.set([0x91], 0);
+        padded.set(namespaceId.reverse(), 1);
+        padded.set(Convert.hexToUint8('00'.repeat(16)), 9);
+        return padded;
+    }
+
+    /**
+     * Converts a decoded address to an encoded address string.
+     * @param {Uint8Array} decoded The decoded address.
+     * @returns {string} The encoded address string corresponding to the input.
+     */
+    public static addressToString = (decoded) => {
+        if (RawAddress.constants.sizes.addressDecoded !== decoded.length) {
+            throw Error(`${Convert.uint8ToHex(decoded)} does not represent a valid decoded address`);
+        }
+
+        return Base32.Base32Encode(decoded);
+    }
+
+    /**
+     * Converts a public key to a decoded address for a specific network.
+     * @param {module:crypto/keyPair~PublicKey} publicKey The public key.
+     * @param {number} networkIdentifier The network identifier.
+     * @returns {Uint8Array} The decoded address corresponding to the inputs.
```

Comment:
UInt8Array return type seems off.

with implementation of `RawAddress` you could return a RawAddress there, which will permit UInt8Array => decoded address.